allennlp/modules/span_extractors/bidirectional_endpoint_span_extractor.py

```diff
@@ -0,0 +1,230 @@
+                 input_dim: int,
+                 forward_combination: str = "y-x",
+                 backward_combination: str = "x-y",
+                 num_width_embeddings: int = None,
+                 span_width_embedding_dim: int = None,
+                 bucket_widths: bool = False,
+                 use_sentinels: bool = True) -> None:
+        super().__init__()
+        self._input_dim = input_dim
+        self._forward_combination = forward_combination
+        self._backward_combination = backward_combination
+        self._num_width_embeddings = num_width_embeddings
+        self._bucket_widths = bucket_widths
+
+        if self._input_dim % 2 != 0:
+            raise ConfigurationError("The input dimension is not divisible by 2, but the "
+                                     "BidirectionalEndpointSpanExtractor assumes the embedded representation "
+                                     "is bidirectional (and hence divisible by 2).")
+        if num_width_embeddings is not None and span_width_embedding_dim is not None:
+            self._span_width_embedding = Embedding(num_width_embeddings, span_width_embedding_dim)
+        elif not all([num_width_embeddings is None, span_width_embedding_dim is None]):
+            raise ConfigurationError("To use a span width embedding representation, you must"
+                                     "specify both num_width_buckets and span_width_embedding_dim.")
+        else:
+            self._span_width_embedding = None
+
+        self._use_sentinels = use_sentinels
+        if use_sentinels:
+            self._start_sentinel = Parameter(torch.randn([1, 1, int(input_dim / 2)]))
+            self._end_sentinel = Parameter(torch.randn([1, 1, int(input_dim / 2)]))
+
+    def get_input_dim(self) -> int:
+        return self._input_dim
+
+    def get_output_dim(self) -> int:
+        unidirectional_dim = int(self._input_dim / 2)
+        forward_combined_dim = get_combined_dim(self._forward_combination,
+                                                [unidirectional_dim, unidirectional_dim])
+        backward_combined_dim = get_combined_dim(self._backward_combination,
+                                                 [unidirectional_dim, unidirectional_dim])
+        if self._span_width_embedding is not None:
+            return forward_combined_dim + backward_combined_dim + \
+                   self._span_width_embedding.get_output_dim()
+        return forward_combined_dim + backward_combined_dim
+
+    @overrides
+    def forward(self, # pylint: disable=arguments-differ
+                sequence_tensor: torch.FloatTensor,
+                indices: torch.LongTensor,
+                sequence_mask: torch.LongTensor = None,
+                indices_mask: torch.LongTensor = None) -> torch.FloatTensor:
+
+        # Both of shape (batch_size, sequence_length, embedding_size / 2)
+        forward_sequence, backward_sequence = torch.split(sequence_tensor, int(self._input_dim / 2), -1)
```

Comment:
You give `dim=-1` below, where you just give `-1` here.  Might be nice to the reader to give the keyword argument every time.  You could also just do `sequence_tensor.split(...)` instead of `torch.split(sequence_tensor, ...)`.